### PR TITLE
fix(types): deduplicate error typings

### LINKED ISSUE #14000: Type errors in database errors code
## Issue Creation Checklist

[x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### SSCCE

**Here is the link to the SSCCE for this issue:** N.A.

### What do you expect to happen?

Typescript compilation to work smoothly.

### What is actually happening?

There are some type errors from `sequelize` codebase:

```bash
node_modules/sequelize/lib/errors/database/exclusion-constraint-error.ts:23:5 - error TS2322: Type '(DatabaseErrorSubclassOptions & ExclusionConstraintErrorOptions) | {}' is not assignable to type 'DatabaseErrorSubclassOptions & ExclusionConstraintErrorOptions'.
  Type '{}' is not assignable to type 'DatabaseErrorSubclassOptions & ExclusionConstraintErrorOptions'.
    Type '{}' is missing the following properties from type 'ExclusionConstraintErrorOptions': constraint, fields, table

23     options = options || {};
       ~~~~~~~

node_modules/sequelize/lib/errors/database/foreign-key-constraint-error.ts:29:5 - error TS2322: Type '(ForeignKeyConstraintErrorOptions & DatabaseErrorSubclassOptions) | {}' is not assignable to type 'ForeignKeyConstraintErrorOptions & DatabaseErrorSubclassOptions'.
  Type '{}' is not assignable to type 'ForeignKeyConstraintErrorOptions & DatabaseErrorSubclassOptions'.
    Type '{}' is missing the following properties from type 'ForeignKeyConstraintErrorOptions': table, fields, value, index, reltype

29     options = options || {};
       ~~~~~~~

node_modules/sequelize/lib/errors/database/unknown-constraint-error.ts:23:5 - error TS2322: Type '(UnknownConstraintErrorOptions & DatabaseErrorSubclassOptions) | {}' is not assignable to type 'UnknownConstraintErrorOptions & DatabaseErrorSubclassOptions'.
  Type '{}' is not assignable to type 'UnknownConstraintErrorOptions & DatabaseErrorSubclassOptions'.
    Type '{}' is missing the following properties from type 'UnknownConstraintErrorOptions': constraint, fields, table

23     options = options || {};
       ~~~~~~~


Found 3 errors.
```

### Additional context

Appeared after the version upgrade: `6.7.0` > `6.14.0`

### Environment

- Sequelize version: 6.14.0
- Node.js version: 16.13.1
- TypeScript version: 4.3.5

### How does this problem relate to dialects?

- [x] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s):
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I don't know how to start, I would need guidance.
- [x] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

### PR DESCRIPTION
**Pull request by ephys**
### Pull Request Checklist

- [n/a] Have you added new tests to prevent regressions?
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [n/a] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

Closes #14000, I hope

This PR deduplicates the typing definitions for Error classes, as `types/lib/errors.d.ts` wasn't removed and was shadowing `lib/errors`.
I also fixed issues detected in `lib/errors` once it wasn't shadowed anymore

### CODE REVIEW

**Comment by WikiRik** [comment]
Just to check, this is compatible with the v6 branch right?

**Comment by ephys** [comment]
These files were converted to TS just before we stopped the migration for v6 :) https://github.com/sequelize/sequelize/tree/v6/lib/errors

**Review by WikiRik — APPROVED** [review]
Looks good!

File: lib/errors/aggregate-error.ts [hunk 1]
```diff
@@ -3,10 +3,11 @@ import BaseError from './base-error';
 /**
  * A wrapper for multiple Errors
  *
- * @param errors Array of errors
+ * @param errors The aggregated errors that occurred
  */
 class AggregateError extends BaseError {
-  errors: Array<AggregateError | Error>;
+  /** the aggregated errors that occurred */
```
Comment (andrejleitner) [inline lib/errors/aggregate-error.ts:9]: Doubled comment? (line 6)

**Review by andrejleitner — APPROVED** [review]

File: lib/errors/aggregate-error.ts (hunk 1 above)
Comment (ephys) [reply lib/errors/aggregate-error.ts:9]: The one on line 6 is for the parameter of the constructor, this one is for the field :)

**Pull request merged**